src/main/java/io/vertx/ext/hawkular/VertxHawkularOptions.java

```diff
@@ -410,4 +414,28 @@ public VertxHawkularOptions addDisabledMetricsType(MetricsType metricsType) {
   public boolean isMetricsTypeDisabled(MetricsType metricsType) {
     return this.disabledMetricsTypes.contains(metricsType);
   }
+
+  public VertxHawkularOptions setFeedId(String feedId) {
+    this.feedId = feedId;
+    return this;
+  }
+
+  public String getFeedId() {
+    if (feedId == null) {
+      throw new RuntimeException("feed id is null.");
```

Comment:
Thought again about this. Please don't validate data here. This option class is a pure data object. Let's move validation to the reporter class (where you can throw an IllegalArgumentException in the constructor).